asm/variable/u64.go

```diff
@@ -0,0 +1,83 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package variable // import "go.opentelemetry.io/ebpf-profiler/asm/variable"
+import "sort"
+
+type U64 interface {
+	// Eval compares this U64 value with another U64 for equality or compatibility.
+	// It returns true if the values are considered equal or compatible according to
+	// the type-specific rules:
+	// - For operations (add, mul, xor): checks if operation types and operands match
+	// - For immediate : checks if values are equal, or extracts value into a Variable
+	// - For memory references: checks if segments and addresses match
+	// - For crop operations: checks if sizes and inner values match
+	// - For variables: checks if they are the same or if one is marked as "any"
+	// Each implementation may have additional specific comparison logic.
+	Eval(v U64) bool
```

Comment:
`Eval` is kinda confusing also for me. As it does not really evaluate the value and return it. Perhaps just `Compare` or `Equals` or `MatchExpression`?

Reply:
naming is hard. I like `MatchExpression`, but then we pass an expression as an argument
Renamed to `Match(pattern Expression)`